Svc/CmdSequencer/test/int/test_cmd_sequencer.py

```diff
@@ -0,0 +1,153 @@
+""" test_cmd_sequencer.py:
+
+A set of integration tests to apply to the CmdSequencer app. This is intended to be a reference of integration testing.
+"""
+
+import subprocess
+import time
+import os
+#from enum import Enum
+from pathlib import Path
+
+#from fprime_gds.common.testing_fw import predicates
+#from fprime_gds.common.utils.event_severity import EventSeverity
+
+## regenerate .seq file
+
+def regenerate_file(seqfile1,deployment):
+    # ex: test_seq.seq
+    oldfile1 = open (seqfile1 +".seq", "r")
+    newfile1 = open (seqfile1 +"_new.seq", "w")
+
+    eachline1 = oldfile1.readlines()
+    line = 1
+    for writing in eachline1:
+        
+        index = writing.find("cmdDisp.")
+        if index != -1:
+            writing = writing[:index] + deployment + "." + writing[index:]
+        
+        newfile1.write(writing)        
+        line = line + 1
+
+    oldfile1.close()
+    newfile1.close()
+    return seqfile1 +"_new.seq"
+
+def test_seqgen(fprime_test_api):
+    
+    """Tests the seqgen can be dispatched (requires localhost testing)"""
+
+    deployment =  fprime_test_api.get_deployment()
+    
+    fprime_test_api.send_and_assert_command(fprime_test_api.getCmdDispName('Svc.CmdSequencer') + '.' + 'CS_AUTO', max_delay=1)
+    
+#    sequence = Path(__file__).parent / "test_seq_new.seq"
+#    sequence2 = Path(__file__).parent / "test_seq_wait_new.seq"
+
+    sequence = Path(__file__).parent / regenerate_file("test_seq",deployment)
+    sequence2 = Path(__file__).parent / regenerate_file("test_seq_wait",deployment)
+
+    
+    assert (
+        subprocess.run(
+            [
+                "fprime-seqgen",
+                "-d",
+                str(fprime_test_api.pipeline.dictionary_path),
+                str(sequence),
+                "/tmp/ref_test_seq.bin",
+            ]
+        ).returncode
+        == 0
+    ), "Failed to run fprime-seqgen"
+
+    assert (
+        subprocess.run(
+            [
+                "fprime-seqgen",
+                "-d",
+                str(fprime_test_api.pipeline.dictionary_path),
+                str(sequence2),
+                "/tmp/ref_test_seq_wait.bin",
+            ]
+        ).returncode
+        == 0
+    ), "Failed to run fprime-seqgen"
+    
+    fprime_test_api.send_and_assert_command(
+        fprime_test_api.getCmdDispName('Svc.CmdSequencer') + '.' + 'CS_RUN', args=["/tmp/ref_test_seq.bin", "BLOCK"], max_delay=5
+    )
+
+    ######    ###### remove newfile1   ######
+    print(f" file:",sequence)
+    os.remove(sequence)
+    os.remove(sequence2)
+    
+def test_send_seq(fprime_test_api):
+    """
+     CS_RUN
+     CS_VALIDATE
+     CS_MANUAL
+     CS_AUTO
+     CS_START
+     CS_CANCEL
+     CS_STEP
+     CS_JOINT_WAIT  (Need to have more 1 run sequence.  Ref don't have more one sequence)
+    """
+
+    fprime_test_api.send_and_assert_command(fprime_test_api.getCmdDispName('Svc.CmdSequencer') + '.' + 'CS_VALIDATE',["/tmp/ref_test_seq.bin"], max_delay=1)    
```

Comment:
This test looks good. But should not use hardcoded files. Topic to discuss with LeStarch